### PR TITLE
feat: Setup barebone CLI

### LINKED ISSUE #141: Integrate Logging
Standarize logging using a log library like https://github.com/Sirupsen/logrus

**Acceptance Criteria**
- [x] Integrate logrus into cobra

**Extra Help**
- https://le-gall.bzh/post/go/integrating-logrus-with-cobra/
- https://ovh.github.io/tat/sdk/golang-full-example/

### LINKED ISSUE #154: Move debug log level into Cobra initi
## This is temporary. Remove it once Cobra/Viper is configured, to handle it as a boolean flags

https://github.com/romie-gr/romie/blob/3f645afa60eb0e660ac709c51da52cd6da653f4c/internal/utils/init.go#L16-L19

---

###### This comment was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 3f645afa60eb0e660ac709c51da52cd6da653f4c in #153. cc @romie-gr.

_Originally posted by @todo[bot] in https://github.com/romie-gr/romie/issues/153#issuecomment-731741087_

### PR DESCRIPTION
**Pull request by drpaneas**
* Add commands: help, version, update, install, remove, search
* Add flags: --config, --debug, --help

fixes #154
fixes #141

### CODE REVIEW

**Comment by zoispag** [comment]
Does this also closes #141?

File: cmd/romie.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
```
Comment (zoispag) [inline cmd/romie.go]: If we agree on separating named imports from non-named, let's rearrange them as:

```suggestion
	"github.com/sirupsen/logrus"
	"github.com/spf13/cobra"
	"github.com/spf13/viper"

	homedir "github.com/mitchellh/go-homedir"
```

The order I usually prefer to use is, alphabetically sorted inside blocks and blocks order:
1. builtin
2. named builtin
3. external
4. named external
5. local
6. named local

```go
import (
  "fmt"
  "strings"
  "strconv"

  fp "path/filepath"

  "github.com/spf13/cobra"
  "github.com/spf13/viper"

  homedir "github.com/mitchellh/go-homedir"
  log "github.com/sirupsen/logrus"

  "github.com/romie-gr/romie/cmd"

  utilities "github.com/romie-gr/romie/utils"
)

Let's discuss more if you want.

File: cmd/search.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	"github.com/spf13/cobra"
+)
+
+// searchCmd represents the search command
+var searchCmd = &cobra.Command{
+	Use:   "search",
+	Short: "Search game ROMs",
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("search called")
+	},
```
Comment (zoispag) [inline cmd/search.go:15]: I think these are not `fmt`ed.
Could you run `go fmt ./...` and commit the changes?

File: .golangci.yml [hunk 3]
```diff
@@ -16,3 +16,6 @@ linters:
     - nlreturn
     - testpackage
     - gochecknoglobals
+    - gochecknoinits
+    - godot
```
Comment (zoispag) [inline .golangci.yml:20]: `Linter that checks if comments end in a period`
Unbelievable 🤣

File: cmd/romie.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.romie.yaml)")
+	rootCmd.PersistentFlags().BoolVar(&debugFlag, "debug", false, "set verbose logs (developers)")
```
Comment (zoispag) [inline cmd/romie.go]:
```suggestion
	rootCmd.PersistentFlags().BoolVar(&debugFlag, "debug", false, "set verbose logs")
```

Don't agree with the `(developers)` part.
For debugging, like when supporting a bug or question, it's also pretty useful.

**Review by zoispag — DISMISSED** [review]

File: cmd/search.go (hunk 2 above)
Comment (drpaneas) [reply cmd/search.go:15]: I've tried it and there are no changes. Plus, the CI is running this anyway and it would have picked up issues related to it.

Comment (zoispag) [reply cmd/search.go:15]: oh, the `=` signs are aligned, not the `:`. Then nevermind my comment! All good :)

**Review by zoispag — APPROVED** [review]
LGTM 💪

**Review by ge0r — APPROVED** [review]

**Pull request merged**